packer_test/common/check/file_gadgets.go

```diff
@@ -33,3 +34,32 @@ func FileExists(filePath string, isDir bool) Checker {
 		isDir:    isDir,
 	}
 }
+
+type fileInDir struct {
+	filename string
+	dirPath  string
+}
+
+func (fe fileInDir) Check(_, _ string, _ error) error {
+	files, err := os.ReadDir(fe.dirPath)
+	if err != nil {
+		return fmt.Errorf("failed to read dir %q: %s", fe.dirPath, err)
+	}
+
+	pattern := regexp.MustCompile(fe.filename)
+
+	for _, file := range files {
+		if !file.IsDir() && pattern.MatchString(file.Name()) {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("file %q not found in dir %q", fe.filename, fe.dirPath)
+}
+
+func FileInDir(dirPath string, filename string) Checker {
```

Comment:
I'm confused about this one, this seems to me to be very close to the `FileExists` checker, is there a nuance I am missing?